javalib/src/main/scala/java/util/Arrays.scala

```diff
@@ -218,55 +212,143 @@ object Arrays {
+    fillImpl(a, 0, a.length, value, checkIndices = false)
+
+  def fill(a: Array[Long], fromIndex: Int, toIndex: Int, value: Long): Unit =
+    fillImpl(a, fromIndex, toIndex, value)
+
+  def fill(a: Array[Int], value: Int): Unit =
+    fillImpl(a, 0, a.length, value, checkIndices = false)
+
+  def fill(a: Array[Int], fromIndex: Int, toIndex: Int, value: Int): Unit =
+    fillImpl(a, fromIndex, toIndex, value)
+
+  def fill(a: Array[Short], value: Short): Unit =
+    fillImpl(a, 0, a.length, value, checkIndices = false)
+
+  def fill(a: Array[Short], fromIndex: Int, toIndex: Int, value: Short): Unit =
+    fillImpl(a, fromIndex, toIndex, value)
+
+  def fill(a: Array[Char], value: Char): Unit =
+    fillImpl(a, 0, a.length, value, checkIndices = false)
+
+  def fill(a: Array[Char], fromIndex: Int, toIndex: Int, value: Char): Unit =
+    fillImpl(a, fromIndex, toIndex, value)
+
+  def fill(a: Array[Byte], value: Byte): Unit =
+    fillImpl(a, 0, a.length, value, checkIndices = false)
+
+  def fill(a: Array[Byte], fromIndex: Int, toIndex: Int, value: Byte): Unit =
+    fillImpl(a, fromIndex, toIndex, value)
+
+  def fill(a: Array[Boolean], value: Boolean): Unit =
+    fillImpl(a, 0, a.length, value, checkIndices = false)
+
+  def fill(a: Array[Boolean], fromIndex: Int, toIndex: Int, value: Boolean): Unit =
+    fillImpl(a, fromIndex, toIndex, value)
+
+  def fill(a: Array[Double], value: Double): Unit =
+    fillImpl(a, 0, a.length, value, checkIndices = false)
+
+  def fill(a: Array[Double], fromIndex: Int, toIndex: Int, value: Double): Unit =
+    fillImpl(a, fromIndex, toIndex, value)
+
+  def fill(a: Array[Float], value: Float): Unit =
+    fillImpl(a, 0, a.length, value, checkIndices = false)
+
+  def fill(a: Array[Float], fromIndex: Int, toIndex: Int, value: Float): Unit =
+    fillImpl(a, fromIndex, toIndex, value)
+
+  def fill(a: Array[AnyRef], value: AnyRef): Unit =
+    fillImpl(a, 0, a.length, value, checkIndices = false)
+
+  def fill(a: Array[AnyRef], fromIndex: Int, toIndex: Int, value: AnyRef): Unit =
+    fillImpl(a, fromIndex, toIndex, value)
+
+  @inline
+  private def fillImpl[@specialized T](a: Array[T], fromIndex: Int, toIndex: Int,
+      value: T, checkIndices: Boolean = true): Unit = {
+    if (checkIndices)
+      checkRangeIndices(a.length, fromIndex, toIndex)
+    var i = fromIndex
+    while (i != toIndex) {
+      a(i) = value
+      i += 1
+    }
+  }
+
+  def copyOf[T <: AnyRef](original: Array[T], newLength: Int): Array[T] = {
+    copyOfImpl(original, newLength,
+      java.lang.reflect.Array.newInstance(
+        original.getClass.getComponentType, _).asInstanceOf[Array[T]])
+  }
+
+  def copyOf[T <: AnyRef, U <: AnyRef](original: Array[U], newLength: Int, newType: Class[_ <: Array[T]]): Array[T] = {
+    copyOfImpl[AnyRef](original.asInstanceOf[Array[AnyRef]], newLength,
+      java.lang.reflect.Array.newInstance(
+        newType.getComponentType, _).asInstanceOf[Array[AnyRef]]).asInstanceOf[Array[T]]
```

Comment:
This incantation with `Array.newInstance` is repeated several times in this file. It should factored out.

Or, better, have `copyOfImpl` take a `ClassTag` instead of a `newArray: Int => Array[T]` callback, and construct the `ClassTag` with `ClassTag(newType)`.